website/content/docs/configuration/client.mdx

```diff
@@ -602,6 +607,41 @@ client {
   [`reserved.reserved_ports`](#reserved_ports) are also reserved on each host
   network.
 
+### `drain_on_shutdown` Block
+
+The `drain_on_shutdown` block controls the behavior of the client when
+[`leave_on_interrupt`][] or [`leave_on_terminate`][] are set. By default
+`drain_on_shutdown` is not configured and clients will not drain on any
+signal.
+
+If `drain_on_shutdown` is configured, the node will drain itself when receiving
+the appropriate signal (`SIGINT` for `leave_on_interrupt` or `SIGTERM` on
+`leave_on_terminate`). By default this acts similarly to running [`nomad node
+drain -self -no-deadline`][]
+
+Note that even if no deadline is set, your init system may send `SIGKILL` to
+Nomad if the drain takes longer than allowed by the service shutdown. For
+example, when running under Linux with `systemd`, you should adjust the
+[`TimeoutStopSec`][] value in the `nomad.service` unit file to account for
+enough time for the client to drain.
+
+```hcl
+client {
+  drain_on_shutdown {
+    deadline = ""
+    ignore_system_jobs = false
+  }
+}
```

Comment:
```suggestion
client {
  # Either leave_on_interrupt or leave_on_terminate must for this to take effect.
  drain_on_shutdown {
    deadline = ""
    ignore_system_jobs = false
  }
}
```

Or something like this. It took me a bit to realize why my example was not working.

Reply:
Good call